content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/list_infra_providers.rb

```diff
@@ -0,0 +1,38 @@
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Transform
+          module Import
+            class ListInfraProviders
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                values_hash = {}
+                values_hash[''] = '-- select target infrastructure provider from list --'
+
+                managers = @handle.vmdb('ManageIQ_Providers_Redhat_InfraManager').all.select(&:validate_import_vm)
```

Comment:
@mkanoor When writing the specs for this script, how should I approach to mocking this call to `vmdb(ManageIQ_Providers_Redhat_InfraManager)`. I tried looking in other specs and found no example of this..

From what I understand the basic idea is to create the backing AR model, e.g. 

`let(:provider)    { FactoryGirl.create(:ems_redhat) }` 

and then create the "wrapping" 

`let(:svc_model_provider) { MiqAeMethodService::MiqAeServiceManageIQ_Providers_Redhat_InfraManager.find(provider.id) }` 

but I'm not sure how to proceed in this case... (tests are currently failing with `"Service Model not found"`.